molgenis-data-security/src/main/java/org/molgenis/data/security/RepositoryCollectionSecurityDecorator.java

```diff
@@ -0,0 +1,131 @@
+import static org.molgenis.data.security.RepositoryPermissionUtils.getCumulativePermission;
+
+public class RepositoryCollectionSecurityDecorator extends AbstractRepositoryCollectionDecorator
+{
+	private static final Logger LOG = LoggerFactory.getLogger(RepositoryCollectionSecurityDecorator.class);
+
+	private final MutableAclService mutableAclService;
+	private final UserPermissionEvaluator userPermissionEvaluator;
+
+	RepositoryCollectionSecurityDecorator(RepositoryCollection delegateRepositoryCollection,
+			MutableAclService mutableAclService, UserPermissionEvaluator userPermissionEvaluator)
+	{
+		super(delegateRepositoryCollection);
+		this.mutableAclService = requireNonNull(mutableAclService);
+		this.userPermissionEvaluator = requireNonNull(userPermissionEvaluator);
+	}
+
+	@Override
+	public Repository<Entity> createRepository(EntityType entityType)
+	{
+		createAcl(entityType);
+		return super.createRepository(entityType);
+	}
+
+	@Override
+	public void updateRepository(EntityType entityType, EntityType updatedEntityType)
+	{
+		validateRepositoryPermission(entityType, WRITEMETA);
+		updateAcl(entityType, updatedEntityType);
+		super.updateRepository(entityType, updatedEntityType);
+	}
+
+	@Override
+	public void addAttribute(EntityType entityType, Attribute attribute)
+	{
+		validateRepositoryPermission(entityType, WRITEMETA);
+		super.addAttribute(entityType, attribute);
+	}
+
+	@Override
+	public void updateAttribute(EntityType entityType, Attribute attr, Attribute updatedAttr)
+	{
+		validateRepositoryPermission(entityType, WRITEMETA);
+		super.updateAttribute(entityType, attr, updatedAttr);
+	}
+
+	@Override
+	public void deleteAttribute(EntityType entityType, Attribute attr)
+	{
+		validateRepositoryPermission(entityType, WRITEMETA);
+		super.deleteAttribute(entityType, attr);
+	}
+
+	@Override
+	public void deleteRepository(EntityType entityType)
+	{
+		validateRepositoryPermission(entityType, WRITEMETA);
+		super.deleteRepository(entityType);
+		mutableAclService.deleteAcl(new RepositoryIdentity(entityType), true);
+	}
+
+	private void createAcl(EntityType entityType)
+	{
+		MutableAcl acl = mutableAclService.createAcl(new RepositoryIdentity(entityType));
+		Package package_ = entityType.getPackage();
+		if (package_ != null)
+		{
+			ObjectIdentity objectIdentity = new PackageIdentity(package_);
+			try
+			{
+				Acl parentAcl = mutableAclService.readAclById(objectIdentity);
+				acl.setParent(parentAcl);
+			}
+			catch (NotFoundException e)
+			{
+				LOG.error("FIXME SystemEntityTypePersister.persistMetadataMetadata is called before packages exist");
+			}
+		}
+		Sid sid = new PrincipalSid(SecurityUtils.getCurrentUsername());
+		acl.insertAce(acl.getEntries().size(), getCumulativePermission(CREATE), sid, true);
+		mutableAclService.updateAcl(acl);
+	}
+
+	private void updateAcl(EntityType entityType, EntityType updatedEntityType)
+	{
+		Package package_ = entityType.getPackage();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)